core/src/slip10/mod.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! MobileCoin SLIP-0010 / BIP39 Based Key Derivation
+//!
+//! This provides utilities to handle SLIP-0010 key bytes and their relation to
+//! the MobileCoin [`Account`](mc_core::Account) structure, which contains a
+//! pair of Ristretto255 view/spend private scalars.
+// As well as providing traits to create a Slip10Key from entropy and path,
+// along with the canonical method of converting a BIP-39
+// [`Mnemonic`](tiny_bip32::Mnemonic) with a given BIP-32 path into a
+// [`Slip10Key`](Slip10Key) usable within MobileCoin.
+
+use curve25519_dalek::scalar::Scalar;
+use hkdf::Hkdf;
+use sha2::Sha512;
+use zeroize::Zeroize;
+
+use mc_crypto_keys::RistrettoPrivate;
+
+#[cfg(feature = "bip39")]
+pub use bip39::{Language, Mnemonic};
+
+use crate::{
+    account::Account,
+    consts::{COINTYPE_MOBILECOIN, USAGE_BIP44},
+    keys::{RootSpendPrivate, RootViewPrivate},
+};
+
+/// Fetch the BIP39 path for a given account index
+pub const fn wallet_path(account_index: u32) -> [u32; 3] {
+    [
+        0x80000000 | USAGE_BIP44,
+        0x80000000 | COINTYPE_MOBILECOIN,
+        0x80000000 | (account_index & !0x80000000),
+    ]
+}
+
+/// A key derived using SLIP-0010 key derivation
+#[derive(Zeroize)]
+#[zeroize(drop)]
+pub struct Slip10Key([u8; 32]);
+
+/// Access [`Slip10Key`] value as byte slice
+impl AsRef<[u8]> for Slip10Key {
+    fn as_ref(&self) -> &[u8] {
+        &self.0[..]
+    }
+}
+
+#[cfg(feature = "internals")]
+impl Slip10Key {
+    /// Create a SLIP-0010 key from raw Ed25519 private key value
+    pub fn from_raw(raw: [u8; 32]) -> Self {
+        Self(raw)
+    }
+}
+
```

Comment:
this is necessary for [hw support](https://github.com/mobilecoinofficial/ledger-mob/blob/main/core/src/lib.rs#L30), i'll have a look at whether `From<Scalar>` can be equivalent

Reply:
The "hw support" link doesn't seem to work.  I'm not sure there is currently a strong practice of documenting features, so don't have a good suggestion on where to place this for future maintainers.

I'm guessing "internals" was discussed already.  For normal language it seems to be overly generic.  It may be a domain term I'm not familiar with.